frontend/index.js

```diff
@@ -6,23 +6,58 @@ import {
     useBase,
     useRecordById,
     useLoadable,
+    useSettingsButton,
+    useViewport,
     useWatchable,
     Box,
-    Text,
-    TextButton,
     Dialog,
-    Link,
     Heading,
+    Link,
+    Text,
+    TextButton,
 } from '@airtable/blocks/ui';
 
+import {useSettings} from './settings';
+import SettingsForm from './SettingsForm';
+
 // How this block chooses a preview to show:
 //
-// - The user selects a row in grid view.
-// - The block looks in the selected field for a supported URL
+// Without a specified Table & Field:
+//  - The user selects a row in grid view.
+//  - The block looks in the Selected Field for a supported URL
 //   (e.g. https://www.youtube.com/watch?v=KYz2wyBy3kc)
-// - The block uses this URL to construct an embed URL and inserts this URL into an iframe.
-
+//  - The block uses this URL to construct an embed URL and inserts this URL into an iframe.
+//
+// With a Specified Table & Specified Field:
+//
+//  - The user may use "Settings" to set a Specified Table and Specified Field for URL previews.
+//  - The user may use "Settings" to toggle the Specified Table and Specified Field constraint.
+//  - The user selects a row in grid view.
+//  - If the Selected Field in the Active Table match the Specified Field & Specified Table, then:
+//      - The block looks in the Selected Field for a supported URL
+//          (e.g. https://www.youtube.com/watch?v=KYz2wyBy3kc)
+//      - If the block supports this URL, then:
+//          - The block uses this URL to construct an embed URL and inserts this URL into an iframe.
+//      - Else,
+//          - Display: "Select a cell to see a preview, View supported URLs"
+//  - Else,
+//      - If the Active Table does not match the Specified Table, then:
+//          - Display: "Switch to the “[Specified Table]” table to see previews."
+//      - If the Selected Field does match the Specified Field, then:
+//          - Display: "Switch to the “[Specified Field]” field to see previews."
+//
+//
 function UrlPreviewBlock() {
+    const viewport = useViewport();
+    const [isSettingsVisible, setIsSettingsVisible] = useState(false);
+    useSettingsButton(() => {
+        if (!isSettingsVisible) {
+            viewport.enterFullscreenIfPossible();
```

Comment:
Automatically switching to full screen feels a little jarring to me. Unlike some other blocks, I can see and do everything I'm interested in from the default view.

Reply:
That's the pattern used in all examples, with the exception of the `name-quiz-block`